Source/UserManagement/Domain/StaffUser/AggregateRoots/StaffUserManagement.cs

```diff
@@ -0,0 +1,161 @@
+                    HandleAddSystemCoordinator(command);
+                    break;
+
+                default:
+                    //TODO: This should not happen due to InputValidation
+                    break;
+                
+            }
+        }
+
+        public void AddPhoneNumber(AddPhoneNumberToStaffUser command)
+        {
+            Apply(new PhoneNumberAddedToStaffUser(
+                command.StaffUserId, command.PhoneNumber, (int)command.Role
+                ));
+        }
+
+        public void RemovePhoneNumber(RemovePhoneNumberFromStaffUser command)
+        {
+            Apply(new PhoneNumberRemovedFromStaffUser(
+                command.StaffUserId, command.PhoneNumber, (int)command.Role
+                ));
+        }
+
+        private void HandleAddAdmin(AddStaffUser command)
+        {
+            Apply(new AdminAdded(
+                command.StaffUserId, command.FullName,
+                command.DisplayName, command.Email
+                ));
+
+        }
+
+        private void HandleAddDataConsumer(AddStaffUser command)
+        {
+            Apply(new DataConsumerAdded(
+                command.StaffUserId, command.FullName,
+                command.DisplayName, command.Email,
+                command.Location.Longitude, command.Location.Latitude
+                ));
+        }
+        private void HandleAddDataCoordinator(AddStaffUser command)
+        {
+            Apply(new DataCoordinatorAdded(
+                command.StaffUserId, command.FullName, command.DisplayName,
+                command.Email, command.YearOfBirth, (int)command.Sex, command.NationalSociety,
+                (int)command.PreferredLanguage, command.Location.Longitude,
+                command.Location.Latitude
+                ));
+
+            if (command.MobilePhoneNumber != null && command.MobilePhoneNumber.Count > 0)
+            {
+                foreach (var number in command.MobilePhoneNumber)
+                {
+                    Apply(new PhoneNumberAddedToStaffUser(
+                        command.StaffUserId, number,
+                        (int)command.Role
+                    ));
+                }
+            }
+            //TODO: DO the same for NationalSocieties
+        }
+        private void HandleAddDataOwner(AddStaffUser command)
+        {
+            Apply(new DataOwnerAdded(
+                command.StaffUserId, command.FullName, command.DisplayName,
+                command.Email, command.YearOfBirth, (int)command.Sex, command.NationalSociety,
+                (int)command.PreferredLanguage, command.Location.Longitude,
+                command.Location.Latitude, command.Position, command.DutyStation
+                ));
+
+            if (command.MobilePhoneNumber != null && command.MobilePhoneNumber.Count > 0)
+            {
+                foreach (var number in command.MobilePhoneNumber)
+                {
+                    Apply(new PhoneNumberAddedToStaffUser(
```

Comment:
I don't have a problem with an Array on an event.

Reply:
According to @einari it is better to have one event per phone number added (i.e. no arrays or as few as possible). That is at least my understanding from previous discussions